packages/number-field/README.md

```diff
@@ -230,11 +265,103 @@ If the user types a value that is between two steps and blurs the input, the val
+
+### States
+
+#### Invalid
+
+The `invalid` attribute indicates that the number field's value is invalid. When set, appropriate ARIA attributes will be automatically applied.
+
+```html
+<sp-field-label for="invalid">
+    It's one banana, Michael, how much could it cost?
+</sp-field-label>
+<sp-number-field
+    id="invalid"
+    invalid
+    style="width: 400px"
+    value="10"
+    min="0"
+    max="0.3"
+    step="0.01"
+    format-options='{
+        "style": "currency",
+        "currency": "USD",
+        "currencyDisplay": "code",
+        "currencySign": "accounting",
+        "minimumFractionDigits": 2,
+        "maximumFractionDigits": 2
+    }'
+></sp-number-field>
+<sp-help-text variant="negative">
+    Value should be between $0 and $0.3.
+</sp-help-text>
+```
+
+#### Valid
+
+The `valid` attribute indicates that the number field's value is valid.
+
+```html
+<sp-field-label for="valid">
+    It's one banana, Michael, how much could it cost?
+</sp-field-label>
+<sp-number-field
+    id="invalid"
+    valid
+    style="width: 400px"
+    value="0.23"
+    min="0"
+    max="0.3"
+    step="0.01"
+    format-options='{
+        "style": "currency",
+        "currency": "USD",
+        "currencyDisplay": "code",
+        "currencySign": "accounting",
+        "minimumFractionDigits": 2,
+        "maximumFractionDigits": 2
+    }'
+></sp-number-field>
+```
+
+#### Required
+
+The `required` attribute sets the number field to an indeterminate state, visually distinct from both checked and unchecked states.
+
+```html
+<sp-field-label for="required" required>Number of tickets</sp-field-label>
+<sp-number-field id="required" required default="0"></sp-number-field>
+<sp-help-text>Number of tickets is required.</sp-help-text>
+```
+
+#### Disabled
+
+The `disabled` attribute prevents the number field from receiving focus or events. The number field will appear faded.
+
+```html
+<sp-field-label for="disabled" disabled>Number of tickets</sp-field-label>
+<sp-number-field id="disabled" disabled value="0"></sp-number-field>
+```
+
+#### Read-only
+
+number fieldes have a `readonly` attribute for when they’re in the disabled state but still need their labels to be shown. This allows for content to be copied, but not interacted with or changed.
+
+```html
+<sp-field-label for="readonly">Number of tickets</sp-field-label>
+<sp-number-field id="readonly" readonly value="0"></sp-number-field>
+```
+
+### Behaviors
+
+#### Default value
 
 The `<sp-number-field>` component doesn't manage a default value by itself. This means that consumers can set the value of the number-field as an empty string by clearing the input. If we want the number-field to reset to a `default-value` when the user clears the input, we can listen for the `change` event on the number-field component and set its value to the desired `default-value` if the input is empty.
 
-```html-live
+```html
 <sp-field-label for="default">
     Default value of this number field is 42
```

Comment:
I don't think this is something you introduced, but the field label says the default label is 42, and then the value of the `sp-number-field` is 20. Is that correct? Just something I noticed.

Reply:
Nice catch